File: backend/app/api/routes/orchestrator.py
from fastapi import APIRouter
from backend.app.schemas.student import StudentSnapshot, OrchestrationResponse
from backend.app.services.orchestrator_service import OrchestratorService

router = APIRouter(prefix="/orchestrate", tags=["orchestrator"])
service = OrchestratorService()


@router.post("", response_model=OrchestrationResponse)
def orchestrate(snapshot: StudentSnapshot) -> OrchestrationResponse:
    return service.run(snapshot)

File: backend-node/src/routes/orchestrator.js
import { Router } from "express";
import { OrchestratorService } from "../services/orchestratorService.js";
import { StudentSnapshotSchema } from "../schemas/student.js";

const router = Router();
const service = new OrchestratorService();

router.post("/orchestrate", async (req, res, next) => {
  try {
    const parsedData = StudentSnapshotSchema.parse(req.body);
    const result = await service.run(parsedData);
    res.json(result);
  } catch (error) {
    if (error.name === "ZodError") {
      res.status(422).json({ detail: error.errors });
    } else if (error.status) {
      res.status(error.status).json({
        detail: error.detail || error.message,
        ...(error.requires_initial_assessment ? { requires_initial_assessment: true } : {}),
      });
    } else {
      next(error);
    }
  }
});

export default router;
